### PR TITLE
DOC/CI: Fix building docs with --no-api

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Fixes running docs via

     python make.py --no-api

A few notes:

 - The removed check `if os.path.exists(path):`; will incorrectly raise if docs are being rebuilt.
 - The value `dirname` returned by `os.walk` is an absolute path, but the check was treating it as relative.
 - If docs are built with api and then rebuilt with --no-api, we want to exclude everything in the `reference` directory; in particular `reference/api`.

### CODE REVIEW

**Comment by rhshadrach** [comment]
@github-actions pre-commit

**Comment by jreback** [comment]
lgtm.

File: doc/make.py [hunk 1]
```diff
@@ -198,11 +206,6 @@ def _add_redirects(self):
                     # sphinx specific stuff
                     title = "this page"
 
-                if os.path.exists(path):
-                    raise RuntimeError(
-                        f"Redirection would overwrite an existing file: {path}"
-                    )
```
Comment (jorisvandenbossche) [inline doc/make.py:204]: Thanks, I regularly ignored this error .. ;)

**Review by jorisvandenbossche — APPROVED** [review]
Thanks, looks good!

**Pull request merged**